src/main/java/game/map/Map.java

```diff
@@ -77,4 +80,42 @@ public List<Player> getPlayers() {
         return players;
     }
 
-}
+        public static HashMap<Player, List<Country>> assignCountries(List<Player> players, List<Country> countries) {
+            HashMap<Player, List<Country>> playerCountryMap = new HashMap<>();
+
+            // Check if the countries list is empty
+            if (countries.isEmpty()) {
+                System.out.println("No countries available to assign.");
+                return playerCountryMap;
+            }
+
+            // Shuffle the countries
+            Collections.shuffle(countries);
+
+            // Calculate the number of countries each player should have
+            int numCountriesPerPlayer = 5;
```

Comment:
Why do you want to define this as 5?
It should be (no of countries) / (no of players).
If there are any remaining countries (if (no of countries) % (no of players) != 0), then you need to assign each of them to a different player (so that you are assigning all countries to players).

Reply:
completed